Project/Project/Pages/Index.cshtml.cs

```diff
@@ -33,30 +33,21 @@ public IActionResult OnGet(string query)
         {
             if(selectListItems == null || !selectListItems.Any())
             {
+                //I have no idea what to change
+                //The code is very delicated
                 CreateSelectList();
             }
 
             ViewData["Code"] = selectListItems;
 
             if (!string.IsNullOrWhiteSpace(query))
             {
-                if (countryDictionary == null || !countryDictionary.Any())
-                {
-                    countryDictionary = selectListItems.ToDictionary(x => x.Value, x => x.Text);
-                }
-
-                Query = countryDictionary[query];
+                CountryDictionary(query);
```

Comment:
A private methods for this block of code. 
Rule of 30:
If an element consists of more than 30 subelements, it is highly probable that there is a serious problem: a) Methods should not have more than an average of 30 code lines (not counting line spaces and comments). b) A class should contain an average of less than 30 methods, resulting in up to 900 lines of code. c) A package shouldn’t contain more than 30 classes, thus comprising up to 27,000 code lines. d) Subsystems with more than 30 packages should be avoided. Such a subsystem would count up to 900 classes with up to 810,000 lines of code. e) A system with 30 subsystems would thus possess 27,000 classes and 24.3 million code lines.